src/UnderlineNav2/UnderlineNavItem.tsx

```diff
@@ -175,7 +176,8 @@ export const UnderlineNavItem = forwardRef(
             ) : (
               counter !== undefined && (
                 <Box as="span" data-component="counter" sx={counterStyles}>
-                  <CounterLabel>{counter}</CounterLabel>
+                  <CounterLabel aria-hidden="true">{counter}</CounterLabel>
+                  <VisuallyHidden>&nbsp;({counter})</VisuallyHidden>
```

Comment:
non-blocking: Maybe it's just relative to the `UnderlineNav`, but I wonder if we should consider adding this behavior inside the `Counter` component

Reply:
I am not really sure if I can follow you, sorry! 

Do you mean like this 
```
<CounterLabel sx={counterStyles}>
    <Box as="span" data-component="counter" aria-hidden="true">
      {counter}
    </Box>
    <VisuallyHidden>&nbsp;({counter})</VisuallyHidden>
</CounterLabel>
```

If so, the counterStyles doesn't really apply to position the counter itself rather they position the string within the counter label. 

**Wrapper inside the CounterLabel**
<img width="102" alt="Screen Shot 2022-11-09 at 10 48 24 am" src="https://user-images.githubusercontent.com/1446503/200708805-febe7781-e103-4b4f-93de-d6a125ef38de.png">

**wrapper outside the CounterLabel** (matches with the design)
<img width="127" alt="Screen Shot 2022-11-09 at 10 48 32 am" src="https://user-images.githubusercontent.com/1446503/200708808-92c9cbd5-446c-45ed-a01c-da8e66391262.png">

Reply:
My bad 🙇 ! The comment was more into adding all of these wrappers as part of the `Counter` component itself. 

It seems that any other component using it, like [Buttons with counters](https://primer.style/react/Button#counter-component) might benefit from adding the `Box` wrapper with the aria-hidden and the `VisuallyHidden` element.

Reply:
Ahh Okay! Yeah, that is a great idea. `CounterLabel` could definitely use some help. How about I create an issue for that and we can discuss from there? I'm going to merge this pull requests as you approved but let me know if you have any concerns 🙌🏼

Reply:
Love it 👍 ❤️ !!

Reply:
Just dropping [the issue](https://github.com/github/primer/issues/1509) here for reference 🙂